refinery/tool_manager/tests.py

```diff
@@ -1598,12 +1663,236 @@ def test_get_tool_launch_config(self, tool_data_mock):
+
+    def test_analysis_group_numbers_list_dsc_non_collection_workflow(self):
+        has_dataset_collection_input_mock = (
+            self.has_dataset_collection_input_mock_false.start()
+        )
+        self.create_valid_tool(
+            ToolDefinition.WORKFLOW,
+            file_relationships=self.LIST
+        )
+        self.update_nodes()
+        self.tool._create_collection_description()
+
+        self.assertTrue(has_dataset_collection_input_mock.called)
+        for index, file_mapping in \
+                enumerate(self.tool._get_galaxy_file_mapping_list()):
+            self.assertEqual(file_mapping[WorkflowTool.ANALYSIS_GROUP], index)
+
+    def test_analysis_group_numbers_pair_dsc_collection_workflow(self):
+        has_dataset_collection_input_mock = (
+            self.has_dataset_collection_input_mock_true.start()
+        )
+        self.create_valid_tool(
+            ToolDefinition.WORKFLOW,
+            file_relationships=self.PAIR
+        )
+        self.update_nodes()
+        self.tool._create_collection_description()
+        self.assertTrue(has_dataset_collection_input_mock.called)
+
+        for file_mapping in self.tool._get_galaxy_file_mapping_list():
+            self.assertEqual(file_mapping[WorkflowTool.ANALYSIS_GROUP], 0)
+
+    def test_analysis_group_numbers_pair_dsc_non_collection_workflow(self):
+        has_dataset_collection_input_mock = (
+            self.has_dataset_collection_input_mock_false.start()
+        )
+        self.create_valid_tool(
+            ToolDefinition.WORKFLOW,
+            file_relationships=self.PAIR
+        )
+        self.update_nodes()
+        self.tool._create_collection_description()
+
+        self.assertTrue(has_dataset_collection_input_mock.called)
+        for file_mapping in self.tool._get_galaxy_file_mapping_list():
+            self.assertEqual(file_mapping[WorkflowTool.ANALYSIS_GROUP], 0)
+
+    def test_analysis_group_numbers_list_pair_dsc_collection_workflow(self):
+        has_dataset_collection_input_mock = (
+            self.has_dataset_collection_input_mock_true.start()
+        )
+        self.create_valid_tool(
+            ToolDefinition.WORKFLOW,
+            file_relationships=self.LIST_PAIR
+        )
+        self.update_nodes()
+        self.tool._create_collection_description()
+        self.assertTrue(has_dataset_collection_input_mock.called)
+        galaxy_file_mapping_list = self.tool._get_galaxy_file_mapping_list()
+
+        self.assertEqual(
+            galaxy_file_mapping_list[0][WorkflowTool.ANALYSIS_GROUP],
+            0
+        )
+        self.assertEqual(
+            galaxy_file_mapping_list[1][WorkflowTool.ANALYSIS_GROUP],
+            0
+        )
+        self.assertEqual(
+            galaxy_file_mapping_list[2][WorkflowTool.ANALYSIS_GROUP],
+            1
+        )
+        self.assertEqual(
+            galaxy_file_mapping_list[3][WorkflowTool.ANALYSIS_GROUP],
+            1
+        )
```

Comment:
Could this be more concise?
```
self.assertEqual(
  [0,0,1,1],
  [item[WorkflowTool.ANALYSIS_GROUP] for item in galaxy_file_mapping_list])
)
```
I like to test larger structures, if possible: When it fails, you'll have more context around the problem.